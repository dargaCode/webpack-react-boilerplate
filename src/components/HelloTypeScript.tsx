// import * as React from "react";
import React from "react";

interface HelloProps {
  compiler: string;
  framework: string;
}

// 'HelloProps' describes the shape of props.
// State is never set so we use the '{}' type.
export default class HelloTypeScript extends React.Component<HelloProps, {}> {
  render() {
    return (
      <h2>
        Hello from {this.props.compiler} and {this.props.framework}!
      </h2>
    );
  }
}
