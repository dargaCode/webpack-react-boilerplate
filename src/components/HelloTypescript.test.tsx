import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import HelloTypeScript from "./HelloTypeScript";

const props = {
  compiler: "jest",
  framework: "react"
};

describe("<HelloTypeScript/>", () => {
  it("should render the correct text", () => {
    const wrapper = shallow(
      <HelloTypeScript compiler={props.compiler} framework={props.framework} />
    );

    expect(wrapper.find(".hello-typescript-message").text()).toBe(
      `Hello from ${props.compiler} and ${props.framework}!`
    );
  });

  it("should match the snapshot", () => {
    const component = renderer.create(
      <HelloTypeScript compiler={props.compiler} framework={props.framework} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
