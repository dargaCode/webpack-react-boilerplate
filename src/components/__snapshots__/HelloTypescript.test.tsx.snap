// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelloTypeScript/> should match the snapshot 1`] = `
<h2
  className="hello-typescript-message"
>
  Hello from 
  jest
   and 
  react
  !
</h2>
`;
